02_housing/0_housing_microdata.qmd

```diff
@@ -163,55 +223,59 @@ write_csv(county_puma,  here::here("02_housing","data","temp","county_puma.csv")
+           HHINCOME = HHINCOME*ADJUST, # adjusts the HH income values by the Census's 12-month adjustment factor (converts numbers into calendar year dollars)
+           PERWT = PERWT*afact, # the weight of each person is adjusted by the area of the PUMA that falls into a given Place
+           RENTGRS = RENTGRS*ADJUST, # adjusts gross monthly rental cost for rented housing units into cal-year dollars
+           OWNCOST = OWNCOST*ADJUST) # adjusts monthly costs for owner-occupied housing units into cal-year dollars
+  
+  # save as "microdata.csv" 
+  write_csv(acs_clean,  here::here("02_housing","data","temp",paste0("place_microdata_", year,".csv")))
+}
 
-# keep only vars we need
-acs_2022 <- housing_ext_def %>%
-  select(HHWT, ADJUST, STATEFIP, PUMA, GQ, OWNERSHP, OWNCOST, RENT, RENTGRS, HHINCOME,
-         VALUEH, VACANCY, PERNUM, PERWT, EDUC, EDUCD, GRADEATT, EMPSTAT, AGE) %>% 
-  # clean up for matching purposes
-  dplyr::rename("puma" = "PUMA",
-                "statefip" = "STATEFIP") %>% 
-  dplyr::mutate(statefip = sprintf("%0.2d", as.numeric(statefip)),
-                puma = sprintf("%0.5d", as.numeric(puma))) %>% 
-# Merge the microdata PUMAs to places
-  left_join(puma_place_2022, by=c( "statefip", "puma")) %>% 
-  mutate(place_code = str_c(statefip, place)) %>% 
-  distinct()
-# now have 3,781,759 rows
-
-# check distinct number of places - get 536 observations
-acs_2022 %>% distinct(statefip, place) %>% nrow()
-
-# run anti_join to see how many cases on the left did not have a match on the right
-test  <- anti_join(acs_2022, puma_place_2022, by=c("statefip","puma"))
-# 1,842,100 obs from the microdata (makes sense since we limited to only PUMAs that are overlapping with Places of interest)
-
-# Drop any observations with NA or 0 for afact (i.e. there is no place of interest overlapping this PUMA)
-acs2022clean <- acs_2022 %>% 
-  tidylog::filter(!is.na(afact)) %>% 
-# removed 1,839,362 rows (49%), 1,942,397 rows remaining
-  filter(afact > 0)
-# no drops
-
-# Adjust weight to account for PUMA-to-county mapping (those where unique_types > 1).;
-
-# Same adjustments as Kevin:
-acs2022clean <- acs2022clean %>%
-  mutate(HHWT = HHWT*afact, # the weight of each household is adjusted by the area of the PUMA that falls into a given Place
-         HHINCOME = HHINCOME*ADJUST, # adjusts the HH income values by the Census's 12-month adjustment factor (converts numbers into calendar year dollars)
-         PERWT = PERWT*afact, # the weight of each person is adjusted by the area of the PUMA that falls into a given Place
-         RENTGRS = RENTGRS*ADJUST, # adjusts gross monthly rental cost for rented housing units into cal-year dollars
-         OWNCOST = OWNCOST*ADJUST) # adjusts monthly costs for owner-occupied housing units into cal-year dollars
-
-# save as "microdata.csv" 
-write_csv(acs2022clean,  here::here("02_housing","data","temp","2022microdata.csv"))
+# loop over years
+map(place_years, prepare_place_data)
```

Comment:
I think this function may over complicate the process. Could you just add a variable to the ACS extract also called crosswalk_period where you make the variable pre-2022 and 2022 based on the ACS years and then add is as an argument in the left_join with the crosswalk? This way you don't have to iterate the process, the join happens once for all the data.